From 136a488d302274a9e771352eb851a2f123d76419 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:05:56 -0400
Subject: [tools] perf/annotate: Print the evsel name in the stdio output

Message-id: <1410181752-25631-112-git-send-email-jolsa@redhat.com>
Patchwork-id: 91830
O-Subject: [PATCH RHEL7.1 BZ1134356 111/307] perf annotate: Print the evsel name in the stdio output
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 9cdbadceca31bf2ccd9e50fecdc5591d3fb8dae3
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Tue Mar 18 11:50:21 2014 -0300

description
===========
So that when showing multiple events annotations, we can figure out
which is which:

  # perf record -a -e instructions,cycles sleep 1
  [ perf record: Woken up 1 times to write data ]
  [ perf record: Captured and wrote 0.826 MB perf.data (~36078 samples) ]
  # perf evlist
  instructions
  cycles
  # perf annotate intel_idle 2> /dev/null | head -1
   Percent | Source code & Disassembly of vmlinux for instructions
  #
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/annotate.c b/tools/perf/util/annotate.c
index 3aa555f..809b4c5 100644
--- a/tools/perf/util/annotate.c
+++ b/tools/perf/util/annotate.c
@@ -1236,6 +1236,7 @@ int symbol__annotate_printf(struct symbol *sym, struct map *map,
  struct dso *dso = map->dso;
  char *filename;
  const char *d_filename;
+ const char *evsel_name = perf_evsel__name(evsel);
  struct annotation *notes = symbol__annotation(sym);
  struct disasm_line *pos, *queue = NULL;
  u64 start = map__rip_2objdump(map, sym->start);
@@ -1243,7 +1244,7 @@ int symbol__annotate_printf(struct symbol *sym, struct map *map,
  int more = 0;
  u64 len;
  int width = 8;
- int namelen;
+ int namelen, evsel_name_len, graph_dotted_len;
 
  filename = strdup(dso->long_name);
  if (!filename)
@@ -1256,14 +1257,17 @@ int symbol__annotate_printf(struct symbol *sym, struct map *map,
 
  len = symbol__size(sym);
  namelen = strlen(d_filename);
+ evsel_name_len = strlen(evsel_name);
 
  if (perf_evsel__is_group_event(evsel))
   width *= evsel->nr_members;
 
- printf(" %-*.*s| Source code & Disassembly of %s\n",
-        width, width, "Percent", d_filename);
- printf("-%-*.*s-------------------------------------\n",
-        width+namelen, width+namelen, graph_dotted_line);
+ printf(" %-*.*s| Source code & Disassembly of %s for %s\n",
+        width, width, "Percent", d_filename, evsel_name);
+
+ graph_dotted_len = width + namelen + evsel_name_len;
+ printf("-%-*.*s-----------------------------------------\n",
+        graph_dotted_len, graph_dotted_len, graph_dotted_line);
 
  if (verbose)
   symbol__annotate_hits(sym, evsel);
-- 
1.7.1